src/components/StatusBadge/index.tsx

```diff
@@ -87,80 +88,77 @@ const StatusBadge = ({
-                      globalMessages.partiallyavailable
-                    ),
-                  }
-                )}
-              </span>
-              {inProgress && <Spinner className="ml-1 h-3 w-3" />}
-            </div>
-          </Badge>
-        </Tooltip>
+      selectedBadge = (
+        <Badge badgeType="success" href={mediaLink}>
+          <div className="flex items-center">
+            <span>
+              {intl.formatMessage(is4k ? messages.status4k : messages.status, {
+                status: intl.formatMessage(globalMessages.partiallyavailable),
+              })}
+            </span>
+            {inProgress && <Spinner className="ml-1 h-3 w-3" />}
+          </div>
+        </Badge>
       );
+      break;
 
     case MediaStatus.PROCESSING:
-      return (
-        <Tooltip content={mediaLinkDescription}>
-          <Badge badgeType="primary" href={mediaLink}>
-            <div className="flex items-center">
-              <span>
-                {intl.formatMessage(
-                  is4k ? messages.status4k : messages.status,
-                  {
-                    status: inProgress
-                      ? intl.formatMessage(globalMessages.processing)
-                      : intl.formatMessage(globalMessages.requested),
-                  }
-                )}
-              </span>
-              {inProgress && <Spinner className="ml-1 h-3 w-3" />}
-            </div>
-          </Badge>
-        </Tooltip>
+      selectedBadge = (
+        <Badge badgeType="primary" href={mediaLink}>
+          <div className="flex items-center">
+            <span>
+              {intl.formatMessage(is4k ? messages.status4k : messages.status, {
+                status: inProgress
+                  ? intl.formatMessage(globalMessages.processing)
+                  : intl.formatMessage(globalMessages.requested),
+              })}
+            </span>
+            {inProgress && <Spinner className="ml-1 h-3 w-3" />}
+          </div>
+        </Badge>
       );
+      break;
 
     case MediaStatus.PENDING:
-      return (
-        <Tooltip content={mediaLinkDescription}>
-          <Badge badgeType="warning" href={mediaLink}>
-            {intl.formatMessage(is4k ? messages.status4k : messages.status, {
-              status: intl.formatMessage(globalMessages.pending),
-            })}
-          </Badge>
-        </Tooltip>
+      selectedBadge = (
+        <Badge badgeType="warning" href={mediaLink}>
+          {intl.formatMessage(is4k ? messages.status4k : messages.status, {
+            status: intl.formatMessage(globalMessages.pending),
+          })}
+        </Badge>
       );
+      break;
 
     default:
-      return null;
+      selectedBadge = null;
+      break;
   }
+
+  // regardless of whether badge should have a tooltip
+  if (selectedBadge === null) {
+    return null;
+  }
+
+  return mediaLink ? (
+    <Tooltip content={mediaLinkDescription}>{selectedBadge}</Tooltip>
+  ) : (
+    selectedBadge
+  );
```

Comment:
These changes should be reverted.

Reply:
(Line 43 above as well.)

Reply:
If I don't remove the `Tooltip` element, there's an empty tooltip that does appear on hover:
![image](https://user-images.githubusercontent.com/20923978/193478039-4e8a4dbb-dd73-441b-a07e-2842afb33499.png)

So, either I refactored the Tooltip component, or I conditionally wrapped the badge in the StatusBadge component, and I chose the later.

Reply:
I think that it's better for the tooltip to not display when empty.

Reply:
In that case, is there a specific way to do this that you have in mind or does what I have here work for you?

Reply:
Ah sorry I meant the Tooltip component should not display anything when content is empty. The logic doesn't belong here in the StatusBadge component IMO.

Reply:
Gotcha, will refactor 👍🏼